src/gtkutils.c

```diff
@@ -35,50 +35,71 @@ GtkWidget *ut_window_new(gchar *title, gchar *wmname, gchar *wmclass,
 }
 
 
-GtkWidget *ut_check_button_new(gchar *label, gboolean active, GtkWidget *parent)
+GtkWidget * gtkutil_dialog_new (char * title,
+                                GtkWindow * parent,
+                                gboolean resizable,
+                                GtkWidget ** main_vbox) /* out */
 {
-   GtkWidget *button;
+   GtkWidget * dialog;
+   dialog = gtk_dialog_new ();
+   gtk_window_set_title (GTK_WINDOW (dialog), title);
+   gtk_container_set_border_width (GTK_CONTAINER (dialog), 4); /* padding */
+   if (parent) {
+      gtk_window_set_transient_for (GTK_WINDOW (dialog), GTK_WINDOW (parent));
+      gtk_window_set_modal (GTK_WINDOW (dialog), TRUE);
+      gtk_window_set_destroy_with_parent (GTK_WINDOW (dialog), TRUE);
+      gtk_window_set_position (GTK_WINDOW (dialog), GTK_WIN_POS_CENTER_ON_PARENT);
+      gtk_window_set_skip_pager_hint (GTK_WINDOW (dialog), TRUE);
+      gtk_window_set_skip_taskbar_hint (GTK_WINDOW (dialog), TRUE);
+   }
+   if (!resizable) {
+      // no need to call this if TRUE, unexpected behavior in GTK3 IIRC
+      gtk_window_set_resizable (GTK_WINDOW (dialog), FALSE);
+   }
 
-   button = gtk_check_button_new_with_label(label);
-   gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(button), active);
-   gtk_box_pack_start(GTK_BOX(parent), button, FALSE, FALSE, 0);
+   if (main_vbox) {
+      *main_vbox = gtk_dialog_get_content_area (GTK_DIALOG (dialog));
+      /* padding */
+      gtk_container_set_border_width (GTK_CONTAINER (*main_vbox), 4);
+   }
 
-   return button;
+   return (dialog);
 }
 
 
-GtkWidget *ut_button_new(gchar *label, gpointer func, gpointer func_data, GtkWidget *parent)
+GtkWidget * gtkutil_frame_vbox (char * label, GtkWidget * parent_box)
 {
-   GtkWidget *button;
+   GtkWidget * frame;
+   frame = gtk_frame_new (label);
+   gtk_box_pack_start (GTK_BOX (parent_box), frame, FALSE, FALSE, 0);
 
-   button = gtk_button_new_with_label(label);
-   g_signal_connect(G_OBJECT(button), "clicked", func, func_data);
-   gtk_box_pack_start(GTK_BOX(parent), button, TRUE, TRUE, 0);
-   gtk_widget_set_can_default (button, TRUE);
+   GtkWidget * frame_vbox = gtk_vbox_new (FALSE, 5);
+   gtk_container_add (GTK_CONTAINER (frame), frame_vbox);
+   /* padding */
+   gtk_container_set_border_width (GTK_CONTAINER (frame_vbox), 5);
 
-   return button;
+   return (frame_vbox);
```

Comment:
Yes, I've double-checked and `return` is a statement in C, hence it doesn't require parens around return value. I do not insist, but I like when there're no extra unneeded parens in the code. It makes easier to distinguish statements vs function calls, for example.